### ISSUE TITLE
Consider using some sort of webhook to monitor for crate uploads

### ISSUE THREAD

**Issue by nagisa** [onur/cratesfyi#14]
Corresponding crates.io issue: https://github.com/rust-lang/crates.io/issues/381

**Comment by Kixiron** [comment]
Triage: I've messed around with the GitHub apu some, I think this could be a viable thing to do because if done right we could just treat github updates as another web request. It'd allow us to respond to crate uploads in a shorter span of time as well as making less requests to github, which is just considerate.

Maybe implement this after the hyper transition, I think it'd be best fit in an async world and would make the transition itself less work

**Comment by jyn514** [comment]
@Kixiron This refers to uploads to crates.io, not anything on github. We'd need crates.io to add a webhook API first: https://github.com/rust-lang/crates.io/issues/381

**Comment by Kixiron** [comment]
Using a github hook would have a similar effect until the crates.io one was implemented, as github would send post requests to us whenever the index repo was updated. I do agree though, using crates.io themselves would be a preferable situation

**Comment by pietroalbini** [comment]
Yeah, we can definitely use the GitHub webhook on the repository index. I can set it up when we are ready.
